tests/test_json_schema.py

```diff
@@ -2692,73 +2683,202 @@ class Model(BaseModel):
         pet: Union[Cat, Dog, Lizard] = Field(..., discriminator='pet_type')
 
     assert Model.model_json_schema() == {
+        '$defs': {
+            'Cat': {
+                'properties': {'pet_type': {'const': 'cat', 'title': 'Pet Type'}},
+                'required': ['pet_type'],
+                'title': 'Cat',
+                'type': 'object',
+            },
+            'Dog': {
+                'properties': {'pet_type': {'const': 'dog', 'title': 'Pet Type'}},
+                'required': ['pet_type'],
+                'title': 'Dog',
+                'type': 'object',
+            },
+            'Lizard': {
+                'properties': {'pet_type': {'enum': ['reptile', 'lizard'], 'title': 'Pet Type'}},
+                'required': ['pet_type'],
+                'title': 'Lizard',
+                'type': 'object',
+            },
+        },
+        'properties': {
+            'pet': {
+                'discriminator': {
+                    'mapping': {
+                        'cat': '#/$defs/Cat',
+                        'dog': '#/$defs/Dog',
+                        'lizard': '#/$defs/Lizard',
+                        'reptile': '#/$defs/Lizard',
+                    },
+                    'propertyName': 'pet_type',
+                },
+                'oneOf': [
+                    {'$ref': '#/$defs/Cat'},
+                    {'$ref': '#/$defs/Dog'},
+                    {'$ref': '#/$defs/Lizard'},
+                    {'$ref': '#/$defs/Lizard'},
```

Comment:
You have `Lizard` twice here, I guess because of the `lizard`/`reptile` business. This is going to fail then anytime someone has a lizard -- because anytime it matches one of those, it will match the second too (and `oneOf` allows only one match). I'm not sure what the intended output is supposed to be, whether you just want to deduplicate before emitting this `oneOf` or if this indeed is meant to always fail.

Reply:
Good catch, I think I figured out how to resolve this